src/bridgeService.ts

```diff
@@ -270,14 +270,11 @@ export class BridgeService {
    */
   public saveCachedPlatformAccessoriesOnDisk(): void {
     try {
-      if (this.cachedPlatformAccessories.length > 0) {
-        this.cachedAccessoriesFileCreated = true;
-
+      // only save the cache file back to disk if we have already attempted to load it
+      // this should prevent the cache being deleted should homebridge be shutdown before it has finished launching
+      if (this.cachedAccessoriesFileLoaded) {
         const serializedAccessories = this.cachedPlatformAccessories.map(accessory => PlatformAccessory.serialize(accessory));
         this.storageService.setItemSync(this.bridgeOptions.cachedAccessoriesItemName, serializedAccessories);
-      } else if (this.cachedAccessoriesFileCreated) {
-        this.cachedAccessoriesFileCreated = false;
-        this.storageService.removeItemSync(this.bridgeOptions.cachedAccessoriesItemName);
```

Comment:
This is the main offending code. If `cachedAccessoriesFileCreated` was not true (as we had not created or loaded the cache yet) the file would be deleted on SIGTERM/SIGINT.